bgpd/bgpd.c

```diff
@@ -2431,8 +2431,13 @@ static void peer_group2peer_config_copy(struct peer_group *group,
 	if (!CHECK_FLAG(peer->flags_override, PEER_FLAG_LOCAL_AS))
 		peer->change_local_as = conf->change_local_as;
 
-	/* TTL */
-	peer->ttl = conf->ttl;
+	/* TTL ... Retain the Peer's TTL if it is not configured in the peer group.  It'll be over-ridden later. */
+	if ((conf->ttl == BGP_DEFAULT_TTL) && (peer->ttl != BGP_DEFAULT_TTL)) {
```

Comment:
The intention here is to set the peer->ttl to conf->ttl only if conf->ttl is configured. i.e. not equal to "BGP_DEFAULT_TTL". Is this right ? If so just checking conf->ttl against BGP_DEFAULT_TTL and setting peer->ttl accordingly is enough. Why do we need the 2nd condition in the if check ? i.e. (peer->ttl != BGP_DEFAULT_TTL)